builtins/util.m4

```diff
@@ -1533,6 +1533,41 @@ ret <16 x double> %ret
 '
 )
 
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; truncate()
+;; Implements uniform and varying trunc() for float and double types.
+;; This uses generic LLVM intrinsics.
+;;
+;; Takes no parameters
+
+define(`truncate', `
+declare float @llvm.trunc.f32(float %val)
+define float @__trunc_uniform_float(float %val) {
+  %r = call float @llvm.trunc.f32(float %val)
+  ret float %r
+}
+
+declare <WIDTH x float> @llvm.trunc.v`'WIDTH`'f32(<WIDTH x float> %val)
+define <WIDTH x float> @__trunc_varying_float(<WIDTH x float> %val) {
+  %r = call <WIDTH x float> @llvm.trunc.v`'WIDTH`'f32(<WIDTH x float> %val)
+  ret <WIDTH x float> %r
+}
+
+declare double @llvm.trunc.f64(double %val)
+define double @__trunc_uniform_double(double %val) {
+  %r = call double @llvm.trunc.f64(double %val)
+  ret double %r
+}
+
+declare <WIDTH x double> @llvm.trunc.v`'WIDTH`'f64(<WIDTH x double> %val)
+define <WIDTH x double> @__trunc_varying_double(<WIDTH x double> %val) {
+  %r = call <WIDTH x double> @llvm.trunc.v`'WIDTH`'f64(<WIDTH x double> %val)
+  ret <WIDTH x double> %r
+}
+')
+
+truncate()
```

Comment:
Even though, instantiating `truncate()` macro unconditionally in the body of util.m4 works, it's the bad way to have it. `ceil/floor/round` are implemented differently (this would be good to refactor and unify, but this is the way it is now) in every builtins file specific for every targets. `truncate()` macro instantiation ideologically need to be next to `ceil/floor/round` implementation.

Would be good to unify `ceil/floor/round` as a separate commit, also. But eventually I see it as calling some `roundings()` macro in all the targets.

Reply:
I'll add the trucate(0 instantiation near floor/ceil